### PR TITLE
WE-574 refactor endpoint exceptions

### PR DESCRIPTION
**Pull request by steinsiv**
## Fixes
This pull request fixes WE-574

## Description
Endpoint exception sends HTTP Header information back to user in GUI. Url needs to be sanitized and b64 encoded text removed.

Additional changes:
- WE-573 refactoring added
- HttpRequestExtension added
- Tests for HttpRequestExtension added

## Type of change

* Bugfix

## Impacted Areas in Application
Login Dialog Frontend

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [x] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [X] I have self-reviewed my code
* [X] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs [hunk 1]
```diff
@@ -40,29 +40,20 @@ public WitsmlClientProvider(IConfiguration configuration, IHttpContextAccessor h
 
             StringValues? authorizationHeader = httpContextAccessor.HttpContext?.Request.Headers["Authorization"];
             StringValues? targetServerHeader = httpContextAccessor.HttpContext?.Request.Headers[WitsmlServerUrlHeader];
-            StringValues? sourceServerHeader = httpContextAccessor.HttpContext?.Request.Headers[WitsmlSourceServerUrlHeader];
 
-            // Use system creds by role from Bearer token
-            if (authorizationHeader?.Count > 0)
+            if (authorizationHeader?.Count > 0 || targetServerHeader?.Count > 0)
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs:46]: Should we throw on unsatisfied if?

File: Src/WitsmlExplorer.Api/Extensions/HttpRequestExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using WitsmlExplorer.Api.Configuration;
+namespace WitsmlExplorer.Api.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+
+        /// The format for WitsmlServer Header value is : "b64(creds)@URL"
+        public static ServerCredentials GetWitsmlServerHttpHeader(this HttpRequest httpRequest, string headerName, Func<string, string> delDecrypt)
+        {
+            string[] headerSplitted = httpRequest.Headers[headerName].ToString().Split('@');
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Extensions/HttpRequestExtensions.cs]: headerSplit? 😄

File: Tests/WitsmlExplorer.Api.Tests/Extensions/HttpRequestExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                Creds = new BasicCredentials()
+                {
+                    UserId = "username",
+                    Password = "password"
+                }
+            };
+        }
+
+        [Fact]
+        public void GetWitsmlServerHttpHeader_ValidInputWCreds_ServerCredentialsCorrectlyParsed()
```
Comment (janburak) [inline Tests/WitsmlExplorer.Api.Tests/Extensions/HttpRequestExtensionsTests.cs]: ValidInputWCreds - does the W stand for With?

File: Src/WitsmlExplorer.Api/Extensions/HttpRequestExtensions.cs (hunk 2 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Api/Extensions/HttpRequestExtensions.cs]: Sure, I need to upgrade my language. I will fix 😀

File: Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs (hunk 1 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs:46]: Yes I agree. What kind of exception would be most suitable in this case you think?

File: Tests/WitsmlExplorer.Api.Tests/Extensions/HttpRequestExtensionsTests.cs (hunk 3 above)
Comment (steinsiv) [reply Tests/WitsmlExplorer.Api.Tests/Extensions/HttpRequestExtensionsTests.cs]: Yes. The method signature was getting wide. But I guess that is a common trait of test methods. 
I have no big preferences, but expanding the `W` to `With` does not make a big impact on length so I can change this

File: Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs (hunk 1 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs:46]: Created a custom one

**Review by janburak — APPROVED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Pull request merged**